src/app/ui/portfolio/utils.ts

```diff
@@ -6,23 +6,68 @@ import {
   parseAsString,
   parseAsStringEnum,
 } from 'nuqs';
-import { map, uniq, uniqBy, flatMap, min, max, sortBy } from 'lodash';
+import { map, uniq, uniqBy, flatMap, min, max, sortBy, sumBy } from 'lodash';
 import type { CacheToken } from 'src/types/portfolio';
 import type {
   PortfolioTokensFilteringParams,
   PortfolioTokensFilter,
   PortfolioDeFiPositionsFilteringParams,
   PortfolioDeFiPositionsFilter,
   SortByEnum,
+  OrderEnum,
 } from './types';
 import type { Account } from '@lifi/wallet-management';
-import type { WalletPositions } from '@/types/jumper-backend';
-import { SortByOptions } from './types';
+import type { DefiPosition, WalletPositions } from '@/types/jumper-backend';
+import { OrderOptions, SortByOptions } from './types';
+
+export interface SortAccessors<T> {
+  getValue: (item: T) => number;
+  getChain: (item: T) => string;
+  getAsset: (item: T) => string;
+}
+
+export const sortPortfolioItems = <T>(
+  items: T[],
+  sortByValue: SortByEnum,
+  order: OrderEnum,
+  accessors: SortAccessors<T>,
+): T[] => {
+  let sorted = [...items];
+
+  if (sortByValue === SortByOptions.VALUE) {
+    sorted = sortBy(sorted, accessors.getValue);
+  } else if (sortByValue === SortByOptions.CHAIN) {
+    sorted = sortBy(sorted, accessors.getChain);
+  } else if (sortByValue === SortByOptions.ASSET) {
+    sorted = sortBy(sorted, accessors.getAsset);
+  }
```

Comment:
It's a bit convoluted, you need to create a structure which maps a name to a sort by option, then you write an if which maps a name to a sort by option, drop one level and make it typesafe.

Something like:

```ts
type SortAccessor<T> = Record<SortBy, (key: T) => string | number>

...

const sorted = sortBy(items, accessors[sortByValue]);

if (order == Desc) {
  return sorted.reverse();
}

return sorted

...

export const tokenSortAccessors: SortAccessors<CacheToken> = {
  SortByOptions.VALUE: (token) => token.cumulatedTotalUSD ?? token.totalPriceUSD ?? 0,
  SortByOptions.CHAIN: (token) => token.chainName ?? '',
  SortByOptions.ASSET: (token) => token.name ?? '',
};
```

Benefit: compiler will tell you when there are missing options, simpler code.